luaclang.c

```diff
@@ -104,6 +106,111 @@ static int cursor_getspelling(lua_State *L)
         return 1;
 }
 
+
+/* Return the cursor kind as a string */
+static const char *cursor_kind_str(enum CXCursorKind kind) 
+{
+    switch (kind) {
+        case CXCursor_StructDecl: 
+                return "StructDecl";
+        case CXCursor_UnionDecl: 
+                return "UnionDecl";
+        case CXCursor_EnumDecl: 
+                return "EnumDecl";
+        case CXCursor_FieldDecl: 
+                return "FieldDecl";
+        case CXCursor_EnumConstantDecl: 
+                return "EnumConstantDecl";
+        case CXCursor_FunctionDecl: 
+                return "FunctionDecl";
+        case CXCursor_VarDecl: 
+                return "VarDecl";
+        case CXCursor_ParmDecl: 
+                return "ParmDecl";
+        case CXCursor_TypedefDecl: 
+                return "TypedefDecl";
+        case CXCursor_IntegerLiteral: 
+                return "IntegerLiteral";
+        default: 
+                return "Unaddressed";
+    }
+}
+
+/*      
+        Format - cur:getKind()
+        Parameter - cur - Cursor whose kind is to be obtained    
+        More info - https://clang.llvm.org/doxygen/group__CINDEX__CURSOR__MANIP.html#ga018aaf60362cb751e517d9f8620d490c
+*/
+static int cursor_getkind(lua_State *L)
+{
+        CXCursor *cur;
+        to_object(L, cur, CURSOR_METATABLE, 1);
+        lua_pushstring(L, cursor_kind_str(clang_getCursorKind(*cur)));
+        return 1;
+}
+
+/* 
+        Visitor function for visit_children
+        A value of this enumeration type should be returned by this functionr to indicate how visit_children proceed : 
```

Comment:
The documentation should be in the function bellow since we don't export this one to Lua.